jwt/api_jws.py

```diff
@@ -95,7 +95,7 @@ def encode(self, payload, key, algorithm='HS256', headers=None,
         signing_input = b'.'.join(segments)
         try:
             alg_obj = self._algorithms[algorithm]
-            key = alg_obj.prepare_key(key)
+            key = alg_obj.prepare_key(key, **kwargs)
```

Comment:
So I'm not sure how I feel about passing `**kwargs` in here. We're basically saying that extra arguments to `encode()` are always destined for `prepare_key()`. What if we had additional arguments that we wanted to be passed to `sign()`?